misc/bazel/c_deps/rust-sys/BUILD.librdkafka.bazel

```diff
@@ -55,6 +56,12 @@ cmake(
         "-DCMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH=0",
         "-DCMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH=0",
         "-DCMAKE_FIND_USE_CMAKE_SYSTEM_PATH=0",
+        # When builing with `CMAKE_BUILD_TYPE=Release`, cmake automatically
+        # sets `-DNDEBUG` flags. This breaks librdkafka, which uses asserts for
+        # runtime error checking (confluentinc/librdkafka#5099). We unset
+        # `NDEBUG` by manually overwriting the cmake's `C_FLAGS` variables.
+        "-DCMAKE_C_FLAGS_RELEASE=-O3",
```

Comment:
We probably also want to include `-fPIC` to compile position independent code.

FWIW if you didn't want to always set `-O3` we could create a `build_mode` flag ([example](https://github.com/MaterializeInc/materialize/blob/main/misc/bazel/platforms/BUILD.bazel#L21-L36)) and here use a `select({ ... })` to use different options for the different modes.

As-is though if this fixes the issue then let's ship it!